abci/client/grpc_client.go

```diff
@@ -314,6 +314,16 @@ func (cli *grpcClient) ApplySnapshotChunkAsync(
 	)
 }
 
+// NOTE: call is synchronous, use ctx to break early if needed
+func (cli *grpcClient) processProposalAsync(ctx context.Context, params types.RequestProcessProposal) (*ReqRes, error) {
```

Comment:
Why do we need this? Can't we just move it into `ProcessProposalSync`?

Reply:
Yeah we can remove this. I originally added it since I thought it was part of the required ABCI method scaffolding, but I don't think it is